src/Output/FreshTrait.php

```diff
@@ -90,7 +90,11 @@ public function isFresh(AssetTarget $target)
 
         $filters = $this->filterRegistry->collection($target);
         foreach ($filters->filters() as $filter) {
-            foreach ($filter->getDependencies($target) as $child) {
+            $dependencies = $filter->getDependencies($target);
+            if ($dependencies === false) {
+                return false;
+            }
```

Comment:
This feels like conflating concerns. Overloading the dependency detection to hand mtime overrides feels wrong. Could we add a new filter hook that allows filters to overrule the mtime value and only implement in the pipe filters.